SPECS/kernel-headers/kernel-headers.spec

```diff
@@ -1,41 +1,82 @@
+# This package doesn't contain any binaries, thus no debuginfo package is needed.
+%global debug_package %{nil}
+
+%if "%{_arch}" == "x86_64"
+    %global build_cross 1
+    %define cross_archs arm64
+%else
+    %global build_cross 0
+    %define cross_archs %{nil}
+%endif
+
 Summary:        Linux API header files
 Name:           kernel-headers
 Version:        5.15.138.1
-Release:        4%{?dist}
+Release:        5%{?dist}
 License:        GPLv2
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Kernel
 URL:            https://github.com/microsoft/CBL-Mariner-Linux-Kernel
-#Source0:       https://github.com/microsoft/CBL-Mariner-Linux-Kernel/archive/rolling-lts/mariner-2/%%{version}.tar.gz
-Source0:        kernel-%{version}.tar.gz
+Source0:        https://github.com/microsoft/CBL-Mariner-Linux-Kernel/archive/rolling-lts/mariner-2/%{version}.tar.gz#/kernel-%{version}.tar.gz
 # Historical name shipped by other distros
 Provides:       glibc-kernheaders = %{version}-%{release}
 BuildArch:      noarch
 
 %description
 The Linux API Headers expose the kernel's API for use by Glibc.
 
+%if %{build_cross}
+%package -n kernel-cross-headers
+Summary: Header files for the Linux kernel for use by cross-glibc.
+
+%description -n kernel-cross-headers
+Kernel-cross-headers includes the C header files that specify the interface
```

Comment:
Can you specify what is different about these headers relative to the standard headers? I can't really tell what this accomplishes for us currently.

Reply:
And why one "cross" package is sufficient when there should really be one for each value in %cross_archs

Reply:
Ah, looks like cross will contain the set of all non-x86 headers.

Reply:
Yes, that's correct. Here I also followed Fedora's approach, which seems reasonable as headers aren't that big and having to use separate ones for each architecture would probably end up being a headache down the road.